tool/scrape_mdn.dart

```diff
@@ -0,0 +1,239 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// ignore_for_file: prefer_expression_function_bodies
+
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:html/dom.dart' as dom;
+import 'package:html/dom_parsing.dart' show TreeVisitor;
+import 'package:html/parser.dart' show parse;
+import 'package:http/http.dart' as http;
+import 'package:pool/pool.dart';
+
+const apiUrl = 'https://developer.mozilla.org/en-US/docs/Web/API';
+
+Future<void> main(List<String> args) async {
+  final client = http.Client();
+
+  // Get the API page with all the interface references.
+  final response = await client.get(Uri.parse(apiUrl));
+  final doc = parse(response.body);
+
+  final section = doc.querySelector('section[aria-labelledby=interfaces]')!;
+  final anchorItems = section.querySelectorAll('li a');
+
+  final interfaceNames = <String>[];
+
+  for (final item in anchorItems) {
+    final href = item.attributes['href']!;
+    final interfaceName = href.split('/').last;
+
+    interfaceNames.add(interfaceName);
+  }
+
+  interfaceNames.sort();
+
+  print('${interfaceNames.length} items read from $apiUrl.');
+
+  final pool = Pool(6);
+
+  final interfaces = await pool.forEach(interfaceNames, (item) async {
+    return populateInterfaceInfo(item, client: client);
+  }).toList();
+
+  client.close();
+
+  const encoder = JsonEncoder.withIndent('  ');
+
+  final file = File('tool/mdn.json');
+  final json = {
+    '__meta__': {
+      'source': '[MDN Web Docs](https://developer.mozilla.org/en-US/docs/Web)',
+      'license':
+          '[CC-BY-SA 2.5](https://creativecommons.org/licenses/by-sa/2.5/)',
+    },
+    for (var i in interfaces) i.name: i.asJson,
+  };
+  file.writeAsStringSync('${encoder.convert(json)}\n');
+}
+
+Future<InterfaceInfo> populateInterfaceInfo(
+  String interfaceName, {
+  required http.Client client,
+}) async {
+  print('  $interfaceName');
```

Comment:
nit: It feels like printing every interface name will likely get too noisy.

Reply:
I'll switch to just printing something once we're done gathering all the info. The whole script takes around 20 seconds to run.